src/System.Net.Primitives/tests/FunctionalTests/Cookie.cs

```diff
@@ -0,0 +1,344 @@
+﻿using System;
+using System.Net;
+
+using Xunit;
+
+namespace System.Net.Primitives.Functional.Tests
+{
+    public static class CookieTest
+    {
+        [Fact]
+        public static void TestConstructors()
+        {
+            Cookie c = new Cookie();
+
+            c = new Cookie("hello", "goodbye");
+            Assert.Equal("hello", c.Name);
+            Assert.Equal("goodbye", c.Value);
+
+            c = new Cookie("hello", "goodbye", "foo");
+            Assert.Equal("hello", c.Name);
+            Assert.Equal("goodbye", c.Value);
+            Assert.Equal("foo", c.Path);
+
+            c = new Cookie("hello", "goodbye", "foo", "bar");
+            Assert.Equal("hello", c.Name);
+            Assert.Equal("goodbye", c.Value);
+            Assert.Equal("foo", c.Path);
+            Assert.Equal("bar", c.Domain);
+        }
+
+        [Fact]
+        public static void TestComment()
+        {
```

Comment:
Regarding naming the test method, i.e `TestComment`...

We don't include the word "test" in the test method name.  We name the test methods according to a 3-part system.  

Usually names of the test methods are X_Y_Z, where X is the property or method name under test, Y is the set of conditions of the test and Z is the expectation.

Please look at other tests in this library or even System.Net.Http/FunctionalTests/... for examples on the naming patterns.

In general, System.Net.* tests use the guidelines as outlined in the book 'The Art of Unit Testing":
http://www.amazon.com/The-Art-Unit-Testing-examples/dp/1617290890/ref=dp_ob_title_bk